extension/extensioncapabilities/go.mod

```diff
@@ -0,0 +1,43 @@
+module go.opentelemetry.io/collector/extension/extensioncapabilities
```

Comment:
Would love a shorter name if we can think of one. I can't, but maybe someone else can

Reply:
Other than just "capabilities" I am out of ideas

Reply:
Could be shortened to "extensioncaps" a la the Linux capabilities family of commands, e.g. https://man7.org/linux/man-pages/man8/getpcaps.8.html. Seems OK as it is though.

Reply:
Yeah, I feel like this is okay, we can rename later if we want to since this is not going 1.0 now. It's definitely on the longer side for a package name but it's not the longest if you take contrib component names into account